server/src/test/java/org/elasticsearch/action/search/ReduceSearchPhaseExceptionTest.java

```diff
@@ -0,0 +1,167 @@
+                + "        \"index\": \"foo\""
+                + "      }"
+                + "    }"
+                + "  ],"
+                + "  \"caused_by\": {"
+                + "                    \"type\": \"runtime_exception\","
+                + "                    \"reason\": null"
+                + "                }"
+                + "}"
+        );
+        System.out.println(Strings.toString(exception));
+        assertEquals(expectedJson, Strings.toString(exception));
+    }
+
+    public void testToAndFromXContent() throws IOException {
+        final XContent xContent = randomFrom(XContentType.values()).xContent();
+
+        ShardSearchFailure[] shardSearchFailures = new ShardSearchFailure[randomIntBetween(1, 5)];
+        for (int i = 0; i < shardSearchFailures.length; i++) {
+            Exception cause = randomFrom(
+                new ParsingException(1, 2, "foobar", null),
+                new InvalidIndexTemplateException("foo", "bar"),
+                new TimestampParsingException("foo", null),
+                new NullPointerException()
+            );
+            shardSearchFailures[i] = new ShardSearchFailure(cause, new SearchShardTarget("node_" + i,
+                new ShardId("test", "_na_", i), null, OriginalIndices.NONE));
+        }
+
+        final String phase = randomFrom("reduce");
+        ReduceSearchPhaseException actual = new ReduceSearchPhaseException(phase, "unexpected failures", new RuntimeException(), shardSearchFailures);
+
+        BytesReference exceptionBytes = toShuffledXContent(actual, xContent.type(), ToXContent.EMPTY_PARAMS, randomBoolean());
+
+        ElasticsearchException parsedException;
+        try (XContentParser parser = createParser(xContent, exceptionBytes)) {
+            assertEquals(XContentParser.Token.START_OBJECT, parser.nextToken());
+            parsedException = ElasticsearchException.fromXContent(parser);
+            assertEquals(XContentParser.Token.END_OBJECT, parser.currentToken());
+            assertNull(parser.nextToken());
+        }
+
+        assertNotNull(parsedException);
+        assertThat(parsedException.getHeaderKeys(), hasSize(0));
+        assertThat(parsedException.getMetadataKeys(), hasSize(1));
+        assertThat(parsedException.getMetadata("es.phase"), hasItem(phase));
+        // SearchPhaseExecutionException has no cause field
+        assertNull(parsedException.getCause().getCause());
+    }
+
+    public void testPhaseFailureWithoutSearchShardFailure() {
+        final ShardSearchFailure[] searchShardFailures = ShardSearchFailure.EMPTY_ARRAY;
+        final String phase = randomFrom("fetch", "search", "other");
+        ReduceSearchPhaseException actual = new ReduceSearchPhaseException(phase, "unexpected failures",
+            new EsRejectedExecutionException("ES rejected execution of fetch phase"), searchShardFailures);
+
+        assertEquals(actual.status(), RestStatus.TOO_MANY_REQUESTS);
```

Comment:
This is not passing since it is making the same assumptions as SearchPhaseExecutionExceptionTests, but you changed "status()", so this needs to be different.